pkg/cgroup/cgroup.go

```diff
@@ -0,0 +1,312 @@
+	if err != nil {
+		return c, failMountCgroup(ver, err)
+	}
+
+	c.mountPath = mountPath
+
+	c.hierarchy, err = GetCgroupControllerHierarchy(DefaultCgroupController)
+	if err != nil {
+		return c, failGetCgroupHierarchy(ver, err)
+	}
+
+	return c, nil
+}
+
+type Cgroup struct {
+	version   CgroupVersion
+	mountPath string
+	hierarchy int
+}
+
+func (c *Cgroup) Destroy() error {
+	err := syscall.Unmount(c.mountPath, 0)
+	if err != nil {
+		return err
+	}
+	err = os.RemoveAll(c.mountPath)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Cgroup) GetMountPoint() string {
+	return c.mountPath
+}
+
+func (c *Cgroup) GetDefaultHierarchy() int {
+	return c.hierarchy
+}
+
+func (c *Cgroup) GetVersion() CgroupVersion {
+	return c.version
+}
+
+//
+// Standalone functions
+//
+
+// GetCgroupPath walks the cgroup fs and provides the cgroup directory path of
+// given cgroupId and subPath (related to cgroup fs root dir). If subPath is
+// empty, then all directories from cgroup fs will be searched for the given
+// cgroupId.
+func GetCgroupPath(rootDir string, cgroupId uint64, subPath string) (string, error) {
+	entries, err := os.ReadDir(rootDir)
+	if err != nil {
+		return "", err
+	}
+
+	for _, entry := range entries {
+		if !entry.IsDir() {
+			continue
+		}
+
+		entryPath := filepath.Join(rootDir, entry.Name())
+		if strings.HasSuffix(entryPath, subPath) {
+			// Lower 32 bits of the cgroup id == inode number of matching cgroupfs entry
+			var stat syscall.Stat_t
+			if err := syscall.Stat(entryPath, &stat); err == nil {
+				// Check if this cgroup path belongs to cgroupId
+				if (stat.Ino & 0xFFFFFFFF) == (cgroupId & 0xFFFFFFFF) {
+					return entryPath, nil
+				}
+			}
+		}
+
+		// No match at this dir level: continue recursively
+		path, err := GetCgroupPath(entryPath, cgroupId, subPath)
+		if err == nil {
+			return path, nil
+		}
+	}
+
+	return "", fs.ErrNotExist
+}
+
+// Returns a cgroup controller hierarchy value
+func GetCgroupControllerHierarchy(subsys string) (int, error) {
+	var value int
+
+	file, err := os.Open(procCgroups)
+	if err != nil {
+		return -1, failCouldNotOpenFile(procCgroups, err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for i := 1; scanner.Scan(); i++ {
+		sline := strings.Fields(scanner.Text())
+		if len(sline) < 2 || sline[0] != subsys {
+			continue
+		}
+		value, err = strconv.Atoi(sline[1])
+		if err != nil || value < 0 {
+			return -1, failErrorParsing(procCgroups, err)
+		}
+	}
+
+	return value, nil
+}
+
+// Checks if cgroupv1 is the default cgroup filesystem
+func IsCgroupV1Default() (bool, error) {
+	var err [3]error
+	var cpu, cpuset, memory int
+
+	cpu, err[0] = GetCgroupControllerHierarchy("cpu")
+	cpuset, err[1] = GetCgroupControllerHierarchy("cpuset")
+	memory, err[2] = GetCgroupControllerHierarchy("memory")
+
+	for _, e := range err {
+		if e != nil {
+			return false, e
+		}
+	}
+
+	if cpu == cpuset && cpu == memory {
```

Comment:
@rafaeldtinoco how can `cpu` be `cpuset` and `memory` at the same time? 
```suggestion
	if cpu == cpuset && cpu == memory {
```

Reply:
Because in cgroupv2 they are all the same Id

Reply:
Ahhh! Maybe add a comment about it?